File: net/OpenDraft.Lang/SymbolToken.cs
// <copyright file="SymbolToken.cs" company="Jon Rowlett">
// Copyright (c) Jon Rowlett. All rights reserved.
// </copyright>

namespace OpenDraft.Lang;

/// <summary>
/// Represents a symbol token in the language.
/// </summary>
public class SymbolToken : Token
{
    /// <summary>
    /// Initializes a new instance of the <see cref="SymbolToken"/> class.
    /// </summary>
    /// <param name="source">The source reference for this token.</param>
    /// <param name="value">The value of the symbol.</param>
    /// <exception cref="ArgumentNullException">Thrown if <paramref name="source"/> is null.</exception>
    public SymbolToken(SourceReference source, Symbol value)
        : base(source)
    {
        this.Value = value;
    }

    /// <summary>
    /// Gets the value of the symbol.
    /// </summary>
    public Symbol Value { get; }
}

File: java/src/io/github/localtoast9001/opendraft/lang/SymbolToken.java
/**
 * Copyright (c) Jon Rowlett. All rights reserved.
 */
package io.github.localtoast9001.opendraft.lang;

/**
 * Represents a symbol token in the OpenDraft language.
 */
public class SymbolToken extends Token {
    private final Symbol symbol;

    public SymbolToken(SourceReference source, Symbol value) {
        super(source);
        if (value == null) {
            throw new IllegalArgumentException("Symbol cannot be null or empty");
        }
        this.symbol = value;
    }

    public Symbol getSymbol() {
        return symbol;
    }

    @Override
    public String toString() {
        return "SymbolToken{" +
                "symbol='" + symbol + '\'' +
                '}';
    }
}
